pkg/kudoctl/cmd/diagnostics/runner_helper.go

```diff
@@ -0,0 +1,167 @@
+	}
+
+	return nil
+}
+
+func runnerForInstance(ir *resourceFuncsConfig, ctx *processingContext) *runner {
+	r := &runner{}
+
+	instance := resourceCollectorGroup{[]resourceCollector{
+		{
+			loadResourceFn: ir.instance,
+			name:           "instance",
+			parentDir:      ctx.operatorDirectory,
+			failOnError:    true,
+			callback:       ctx.setOperatorVersionNameFromInstance,
+			printMode:      ObjectWithDir},
+		{
+			loadResourceFn: ir.operatorVersion(ctx.operatorVersionName),
+			name:           "operatorversion",
+			parentDir:      ctx.operatorDirectory,
+			failOnError:    true,
+			callback:       ctx.setOperatorNameFromOperatorVersion,
+			printMode:      ObjectWithDir},
+		{
+			loadResourceFn: ir.operator(ctx.operatorName),
+			name:           "operator",
+			parentDir:      ctx.rootDirectory,
+			failOnError:    true,
+			printMode:      ObjectWithDir},
+	}}
+	r.addCollector(instance)
+
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.pods,
+		name:           "pod",
+		parentDir:      ctx.instanceDirectory,
+		callback:       ctx.setPods,
+		printMode:      ObjectListWithDirs})
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.services,
+		name:           "service",
+		parentDir:      ctx.instanceDirectory,
+		printMode:      RuntimeObject})
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.deployments,
+		name:           "deployment",
+		parentDir:      ctx.instanceDirectory,
+		printMode:      RuntimeObject})
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.statefulSets,
+		name:           "statefulset",
+		parentDir:      ctx.instanceDirectory,
+		printMode:      RuntimeObject})
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.replicaSets,
+		name:           "replicaset",
+		parentDir:      ctx.instanceDirectory,
+		printMode:      RuntimeObject})
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.statefulSets,
+		name:           "statefulset",
+		parentDir:      ctx.instanceDirectory,
+		printMode:      RuntimeObject})
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.serviceAccounts,
+		name:           "serviceaccount",
+		parentDir:      ctx.instanceDirectory,
+		printMode:      RuntimeObject})
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.clusterRoleBindings,
+		name:           "clusterrolebinding",
+		parentDir:      ctx.instanceDirectory,
+		printMode:      RuntimeObject})
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.roleBindings,
+		name:           "rolebinding",
+		parentDir:      ctx.instanceDirectory,
+		printMode:      RuntimeObject})
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.clusterRoles,
+		name:           "clusterrole",
+		parentDir:      ctx.instanceDirectory,
+		printMode:      RuntimeObject})
+	r.addCollector(&resourceCollector{
+		loadResourceFn: ir.roles,
+		name:           "role",
+		parentDir:      ctx.instanceDirectory,
+		printMode:      RuntimeObject})
+	r.addCollector(&logsCollector{
```

Comment:
This is another reason why a dedicated `PodCollector` might be better: right now the complexity is hidden in the fact that `logCollector` has to be last and it depends on the `ctx.podList`. A `PodCollector` would collect both: `pod.yaml` and `pod.log` as an "atomic" unit.